src/Symfony/Component/Lock/Store/ZookeeperStore.php

```diff
@@ -0,0 +1,183 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Lock\Store;
+
+use Symfony\Component\Lock\Exception\LockAcquiringException;
+use Symfony\Component\Lock\Exception\LockConflictedException;
+use Symfony\Component\Lock\Exception\LockReleasingException;
+use Symfony\Component\Lock\Exception\NotSupportedException;
+use Symfony\Component\Lock\Key;
+use Symfony\Component\Lock\StoreInterface;
+
+/**
+ * ZookeeperStore is a StoreInterface implementation using Zookeeper as store engine.
+ *
+ * @author Ganesh Chandrasekaran <gchandrasekaran@wayfair.com>
+ */
+class ZookeeperStore implements StoreInterface
+{
+    /**
+     * @var \Zookeeper
+     */
+    private $zookeeper;
+
+    /**
+     * Locker constructor.
+     *
+     * @param \Zookeeper $zookeeper Zookeeper Instance
+     */
+    public function __construct(\Zookeeper $zookeeper)
+    {
+        $this->zookeeper = $zookeeper;
+    }
+
+    /**
+     * Stores the resource if it's not locked by someone else.
+     *
+     * @param Key $key the Key that represent the lock resource
+     *
+     * @throws LockConflictedException
+     * @throws LockAcquiringException
+     */
+    public function save(Key $key)
+    {
+        if ($this->exists($key)) {
+            return;
+        }
+
+        $resource = $this->getKeyResource($key);
+        $token = $this->getUniqueToken($key);
+
+        try {
+            $this->createNewLock($resource, $token);
+        } catch (LockConflictedException $ex) {
+            throw $ex;
+        } catch (\Throwable $throwable) {
+            throw new LockAcquiringException($throwable);
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @throws LockReleasingException
```

Comment:
if this is legit, shouldn't it be added to the interface?
throwing new exception is an API break if we are strict, isn't it?
why throw anything at all actually?

Reply:
Yes it should be part of the interface. Fixed. As mentioned in my comment, the client should know when the lock is not released cleanly when calling this function and it should handle the case accordingly.